src/include/osl_pvt.h

```diff
@@ -629,6 +629,18 @@ class Symbol {
         m_lastwrite  = std::max (m_lastwrite, lw);
     }
 
+    // Mark the symbol as always being read (and, if write==true, also
+    // that it's always written). Then is for when we don't know when
+    // it's read or written, but want to be sure it doesn't look unused.
+    void mark_always_used (bool write=false) {
```

Comment:
The important bit here is that I noticed that the old code (in addition to being needlessly repeated) added the symbol but didn't mark it read or written at all. It doesn't seem to lead to any bugs at the moment, but some optimizations do look at when symbols are used, so I feared a source of future bugs. So I wanted to be sure that any time we added a symbol, it at least knew it was used, and with an expansive lifetime, so nothing could go wrong. The lifetime (and whether it reads or writes) will be tightened up at the end of the current pass, when everything is reanalyzed.